mono/mini/mini-runtime.c

```diff
@@ -1905,6 +1905,97 @@ mono_jit_map_is_enabled (void)
 
 #endif
 
+#if ENABLE_JIT_DUMP
+static FILE* perf_dump_file;
+uint64_t codeIndex = 0;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+void *mmapAddr = MAP_FAILED;
+
+void
+mono_enable_jit_dump (void)
+{
+        if (!perf_dump_file) {
+		char name [64];
+		FileHeader * header = malloc (sizeof (FileHeader));
+		int result = 0;
+		
+		result = pthread_mutex_lock (&mutex);
+		
+		g_snprintf (name, sizeof (name), "/tmp/jit-%d.dump", getpid ());
+		unlink (name);
+		perf_dump_file = fopen (name, "w");
+		
+		add_file_header_info (header);
+		fwrite (header, sizeof (FileHeader), 1, perf_dump_file);
+		
+		mmapAddr = mmap(NULL, sizeof (FileHeader), PROT_READ | PROT_EXEC, MAP_PRIVATE, 1, 0);
+		
+		result = pthread_mutex_unlock (&mutex);
+	}
+}
+
+void
+add_file_header_info (FileHeader *header)
+{
+	header->magic = JIT_DUMP_MAGIC;
+	header->version = JIT_DUMP_VERSION;
+	header->total_size = sizeof (FileHeader);
+	header->elf_mach = ELF_MACHINE;
+	header->pad1 = 0;
+	header->pid = getpid ();
+	header->timestamp = getTimeStampNS ();
+	header->flags = 0;
+}
+
+guint64
+getTimeStampNS (void)
```

Comment:
Can you use `g_get_monotonic_time()` instead of defining this function?

Reply:
We don't have that one in `eglib` in Mono.  but there's a version of this same code in `mini-posix.c` in `clock_get_time_ns` that should be made into a utility function in utils/

Reply:
will refactor the code in a follow-up PR.

Reply:
Tracked by dotnet/runtime #34074